src/main/java/com/google/sps/servlets/HistoryServlet.java

```diff
@@ -53,9 +55,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             }
         }
 
-        String json = new Gson().toJson(scheduledSessions);
+        List<TutorSession> previousSessions = new ArrayList<TutorSession>();
+
+        filterPastSessions(scheduledSessions, previousSessions);
+
+        String json = new Gson().toJson(previousSessions);
         response.setContentType("application/json;");
         response.getWriter().println(json);
         return;
     }
+
+    private void filterPastSessions(List<TutorSession> allSessions, List<TutorSession> previousSessions) {
+        Date currentDate = new Date();
+
+        for (TutorSession session : allSessions) {
+            Calendar sessionCalendar = session.getTimeslot().getDate();
+            int start = session.getTimeslot().getStart();
+            int hour = start / 60;
+            int minute = start % 60;
+
+            sessionCalendar.set(Calendar.HOUR_OF_DAY, hour);
+            sessionCalendar.set(Calendar.MINUTE, minute);
+
+            Date sessionDate = sessionCalendar.getTime();
+            int comparison = currentDate.compareTo(sessionDate);
+            if (comparison == 0 || comparison == 1) {
```

Comment:
Since this and the upcoming sessions logic add sessions when the current date is equal to the session date, couldn't a session appear both on the history and upcoming sessions page?

Reply:
Yes, this would be the case if a session time and date equal the current time and date. I think it would be fine for it to be displayed on both upcoming and history since it would mean that the session is currently happening. IN addition, this would only happen for one minute because after that the current time would not be equal to the session time and that session would, thus, only show up on the history page.

Reply:
Oh I see. Thanks for clarifying